addons/account/static/src/js/reconciliation/reconciliation_model.js

```diff
@@ -589,7 +591,20 @@ var StatementModel = BasicModel.extend({
                 "new_aml_dicts": _.map(_.filter(props, function (prop) {
                     return isNaN(prop.id);
                 }), self._formatToProcessReconciliation.bind(self, line)),
-            });
+            };
+            // If the lines are not fully balanced, create an unreconciled amount
+            var currency = session.get_currency(line.st_line.currency_id);
```

Comment:
not sure about the currency you're using here: if the statement line has a currency, you should use it, otherwise you should use the currency of the statement's journal (and if the journal don't have any, it's gonna be the company currency)

Reply:
`line.st_line.currency_id` is computed by
https://github.com/odoo/odoo/blob/saas-16/addons/account/models/account_move.py#L68 . 

I'm not sure about the journal's currency...

Reply:
If I let the currency like this, its equivalent to:
`currency_id or company.currency_id or user.company_id.currency_id`

Is it wrong?

Reply:
`line.st_line.currency_id` is never `false` because:
https://github.com/odoo/odoo/blob/saas-16/addons/account/models/account_bank_statement.py#L525

Reply:
ok then can you remove the currency clause from bellow if statement, and replace that with a comment explaining what value 'currency' might have